internal/machine/ephemeral.go

```diff
@@ -123,7 +123,9 @@ func makeCleanupFunc(ctx context.Context, machine *api.Machine) func() {
 	return func() {
 		const stopTimeout = 15 * time.Second
 
-		stopCtx, cancel := context.WithTimeout(context.Background(), stopTimeout)
+		// FIXME: is there a reason we *need* to use context.Background here, instead of the normal context
+		// As far as I can tell, this is the only place in the codebase that does this
+		stopCtx, cancel := context.WithTimeout(ctx, stopTimeout)
```

Comment:
I need to verify that this is fine (though I haven't experienced any issues due to this change)